core/code/layerchooser.js

```diff
@@ -133,6 +133,47 @@ var LayerChooser = L.Control.Layers.extend({
     return this;
   },
 
+  _onLongClick: function (idx, originalEvent) {
+    var el = this._layers[idx];
+    var defaultPrevented;
+
+    // @section Extra events
+    // Fired when the overlay's label is long-clicked in the layers control.
+    // Besides [standard](https://leafletjs.com/reference.html#layerscontrolevent) properties
+    // contains also following properties:
+
+    // @miniclass LayersControlEventExt
+    // @inherits LayersControlEvent
+    // @property control: L.Control
+    // layerChooser instance.
+    // @property idx: Number
+    // Internal index of overlay.
+    // @property originalEvent: DOMEvent
+    // The original mouse/jQuery event that triggered this Leaflet event.
+    // @property preventDefault: Function
+    // Method to prevent default action of event (like overlays toggling), otherwise handled by layerChooser.
+    var obj = {
+      control: this,
+      idx: idx,
+      originalEvent: originalEvent || {type: 'taphold'},
+      preventDefault: function () {
+        defaultPrevented = true;
+        this.defaultPrevented = true;
+      }
+    };
+
+    // @event longclick: LayersControlEventExt
+    // Fired on layer
+    el.layer.fire('longclick', obj);
+
+    // @event overlaylongclick: LayersControlEventExt
+    // Fired on map
+    this._map.fire('overlaylongclick', L.extend(obj, el));
+    if (!defaultPrevented) {
```

Comment:
why not, but I don't see use case since the default is still toggleOverlay
do you expect an override like:
```js
map.on('overlaylongclick', function (e) { 
  e.preventDefault(); 
  doThings();
});

Reply:
I don't know, will decide tomorrow.

Reply:
I expect something like this:
```js
map.on('overlaylongclick', function (e) {
  if (!e.metaKey) { return; }
  e.preventDefault();
  doThings();
});
// then in different plugin
map.on('overlaylongclick', function (e) {
  if (!e.ctrlKey) { return; }
  e.preventDefault();
  doOtherThings();
});
```
So you are right, explicit `e.preventDefault();` is not convenient, but I have not found better way to detect if default action is still needed after all events processed.